### PR TITLE
Add go-mod-tidy target and action

### PR DESCRIPTION
**Pull request by MrAlias**

### CODE REVIEW

**Review by MikeGoldsmith — APPROVED** [review]

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -46,7 +46,9 @@ jobs:
           path: ~/.tools
           key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('./internal/tools/**') }}
       - name: Run linters
-        run: make license-header-check
+        run: make license-header-check go-mod-tidy
```
Comment (pellared) [inline .github/workflows/checks.yml:49]: How about simplifying to the following?

```suggestion
        run: make precommit
```

**Review by pellared — APPROVED** [review]

File: .github/workflows/checks.yml (hunk 1 above)
Comment (MrAlias) [reply .github/workflows/checks.yml:49]: This is going to be different than `precommit`, see https://github.com/open-telemetry/opentelemetry-go/blob/86f325839ca329997672fbae8b054e7b6181b4d2/.github/workflows/ci.yml#L47

Comment (pellared) [reply .github/workflows/checks.yml:49]: I see. How about encapsulating it to a `ci` Make target?

Comment (pellared) [reply .github/workflows/checks.yml:49]: Make sense. Resolving.

**Pull request merged**